Automattic/jetpack
# Packages: Make logo package tests independent

This PR changes the tests of the logo package to live in the package and run independently. It also updates our CI to run them in separate jobs, and allow for the same to work for other packages.

#### Changes proposed in this Pull Request:
* Move the logo package tests to the logo package.
* Rewrite the tests to use `php-mock`, as it supports >= PHP 5.6.
  * Also tried with WP_mock, but it requires >= PHP 7.1
  * Also tried with php-mock-phpunit but it requires PHP >= 7.0
* Rewrite the tests to be independent of the WP Core Unit Test framework, and independent from the Jetpack tests as well.
* Create a `composer phpunit` composer command to be ran inside the logo package directory. It allows for the tests to be ran independently from Jetpack.
* Updated our Travis CI config so it runs the packages tests in a separate job for each PHP version.
* Gitignore `vendor` dir in each package.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* The PR updates the tests of the logo package and the infrastructure that runs them. Part of the Jetpack DNA project.

#### Testing instructions:
* Checkout this branch.
* Navigate to `packages/logo`
* Run `composer phpunit`.
* Verify that runs `composer install` internally, and then runs the package tests.
* Verify the tests pass.
* Run `composer phpunit` again.
* Verify the second time is fast because the dependencies are already installed.
* Verify CI passes on this PR.

#### Proposed changelog entry for your changes:
* Make logo package tests independent.
